integration/spark/shared/src/main/java/io/openlineage/spark/agent/lifecycle/plan/column/ColumnLevelLineageBuilder.java

```diff
@@ -87,16 +96,29 @@ public void addOutput(ExprId exprId, String attributeName) {
    * @param inputExprId
    */
   public void addDependency(ExprId outputExprId, ExprId inputExprId) {
-    exprDependencies
-        .computeIfAbsent(outputExprId, k -> new HashSet<>())
-        .add(new Dependency(inputExprId, TransformationInfo.identity()));
+    addDependency(outputExprId, inputExprId, TransformationInfo.identity());
   }
 
   public void addDependency(
       ExprId outputExprId, ExprId inputExprId, TransformationInfo transformationInfo) {
-    exprDependencies
-        .computeIfAbsent(outputExprId, k -> new HashSet<>())
-        .add(new Dependency(inputExprId, transformationInfo));
+    dependenciesAdded++;
+
+    if (dependenciesAdded > COMPUTED_DEPENDENCY_HARD_LIMIT) {
+      // do nothing -> hard limit of allowed dependencies reached
+      return;
+    }
```

Comment:
each time addDependency() will be called, the `dependenciesAdded` value will be incremented - it makes no sense, if statement should be first

Reply:
yeah, no need to increment the counter. Although the main point is just to stop adding new dependencies.

Reply:
thx